### ISSUE TITLE
Use project template without frontend

### ISSUE THREAD

**Issue by zaanposni** [jasontaylordev/CleanArchitecture#145]
**Is your feature request related to a problem? Please describe.**
Hi,
I watched your presentation at GOTO2019 and really liked  the setup of your template. However I want to implement an API without any frontend.
Is there a complete guide on how to remove anything related to the frontend?

**Describe the solution you'd like**
Give the option to install a subversion/extension of this template without an implemented frontend (I do not know if this is possible in dotnet)

**Describe alternatives you've considered**
I tried to manually remove everything that looked suspicious to me.
- Anything SPA related in the `src/WebUI/webUI.csproj` 
- The complete `ClientApp/` directory
- This section in `src/WebUI/Startup.cs`
```
app.UseSpa(spa =>
            {
                // To learn more about options for serving an Angular SPA from ASP.NET Core,
                // see https://go.microsoft.com/fwlink/?linkid=864501

                spa.Options.SourcePath = "ClientApp";

                if (env.IsDevelopment())
                {
                    spa.UseAngularCliServer(npmScript: "start");
                }
            });
```

Template compiles successfully after these steps I just do not feel right about my approach.

**Additional context**
Surely it should be enough if you could comment if I did everything correctly so if  someone googles this issue he can reproduce my steps.

**Comment by o-l-i-g** [comment]
I'm on the same journey as you @zaanposni, I wanted to remove dependency to Angular and NSwag and I did the same thing to begin with. I too didn't feel confidence as I don't have experience dealing with SPA or NSwag - so instead I created a separate ASP.NET Core web app project and copy & pasted the relevant aspects for dependency injection (Application/Infrastructure layers and CurrentUserService etc). Now I'm confident I 'removed' dependency to SPA.

Just for completeness to your strategy I wanted to add a few more details in case you missed it.
In the Startup.cs with the current template (near lines 54 and 91) there's these bits too:

`
// In production, the Angular files will be served from this directory
            services.AddSpaStaticFiles(configuration =>
            {
                configuration.RootPath = "ClientApp/dist";
            });
`
and
`
app.UseStaticFiles();
            if (!env.IsDevelopment())
            {
                app.UseSpaStaticFiles();
            }
`
And in the template the WebUI project has a package dependency to Microsoft.AspNetCore.SpaServices.Extensions

And appsettings.json's entry for identity server does have SPA in a assigned to a value but I don't think it's relevant:

`
  "IdentityServer": {
    "Clients": {
      "SolutionName.WebUI": {
        "Profile": "IdentityServerSPA"
      }
    }
  },
`
